WalletWasabi/Tor/Http/TorHttpClient.cs

```diff
@@ -62,10 +67,17 @@ public async Task<HttpResponseMessage> SendAsync(HttpMethod method, string relat
 	}
 
 	/// <exception cref="OperationCanceledException">If <paramref name="token"/> is set.</exception>
-	public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken token = default)
+	public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken token = default)
 	{
-		ICircuit circuit = Mode is Mode.NewCircuitPerRequest ? new OneOffCircuit() : PredefinedCircuit!;
-		return TorHttpPool.SendAsync(request, circuit, token);
+		if (Mode is Mode.NewCircuitPerRequest)
+		{
+			using PersonCircuit circuit = new();
```

Comment:
I use `PersonCircuit` to model `OneOffCircuit`. So the new abstraction seems better to me.